weldx/core.py

```diff
@@ -0,0 +1,492 @@
+            if check_structural_equality:
+                equality = self.expression == other.expression
+            else:
+                equality = sympy.simplify(self.expression - other.expression) == 0
+
+            if check_parameters:
+                equality = equality and self._parameters == other.parameters
+            return equality
+        return False
+
+    def set_parameter(self, name, value):
+        """Define an expression parameter as constant value.
+
+        Parameters
+        ----------
+        name
+            Name of the parameter used in the expression.
+        value
+            Parameter value. This can be number, array or pint.Quantity
+
+        """
+        self._parameters[name] = value
+
+    @property
+    def num_parameters(self):
+        """Get the expressions number of parameters.
+
+        Returns
+        -------
+        int:
+            Number of parameters.
+
+        """
+        return len(self._parameters)
+
+    @property
+    def num_variables(self):
+        """Get the expressions number of free variables.
+
+        Returns
+        -------
+        int:
+            Number of free variables.
+
+        """
+        return len(self.expression.free_symbols) - len(self._parameters)
+
+    @property
+    def expression(self):
+        """Return the internal sympy expression.
+
+        Returns
+        -------
+        Internal sympy expression
+
+        """
+        return self._expression
+
+    @property
+    def parameters(self) -> Dict:
+        """Return the internal parameters dictionary.
+
+        Returns
+        -------
+        Dict
+            Internal parameters dictionary
+
+        """
+        return self._parameters
+
+    def get_variable_names(self):
+        """Get a list of all expression variables.
+
+        Returns
+        -------
+        List
+
+        """
+        variable_names = []
+        for var in self._expression.free_symbols:
+            if var.__str__() not in self._parameters:
+                variable_names.append(var.__str__())
+        return variable_names
+
+    def evaluate(self, **kwargs) -> Any:
+        """Evaluate the expression for specific variable values.
+
+        Parameters
+        ----------
+        kwargs
+            additional keyword arguments (variable assignment) to pass.
+
+        Returns
+        -------
+        Any:
+            Result of the evaluated function
+
+        """
+        intersection = set(kwargs).intersection(self._parameters)
+        if len(intersection) > 0:
+            raise ValueError(
+                f"The variables {intersection} are already defined as parameters."
+            )
+        inputs = {**kwargs, **self._parameters}
+        return self.function(**inputs)
+
+
+# TimeSeries ---------------------------------------------------------------------------
+
+
+class TimeSeries:
+    """Describes the behaviour of a quantity in time."""
+
+    _valid_interpolations = ["step", "linear"]
```

Comment:
I think we can remove this, implement the `step` case and otherwise just let xarray handle (or fail) the interpolation

Reply:
This is only needed for the constructor. Assume someone creates a TimeSeries and stores it to an asdf file. Without this, you can basically store everything there as long as you don't interpolate.